packages/zipkin-instrumentation-koa/index.d.ts

```diff
@@ -0,0 +1,17 @@
+import {Middleware} from 'koa';
+import {Tracer} from 'zipkin';
+
+/**
+ * When a request comes in, creates ServerRecv annotation and then passes
+ * the request to the next middleware. When the final middleware finishes
+ * the request, creates ServerSend annotation
+ *
+ * Sets the tracer.id to the span id prior to running the next middleware.
+ *
+ * Note that if the next middleware makes async calls, it should either
+ * store the span id manually or use a CLSContext so that the annotations
```

Comment:
I think you mean access the span id? Maybe you can mention the sneaky property we add to requests (`_trace_id`) so they know what to read. While we don't do a great job documenting this could be a good place to start.

Also, CLSContext would only work if the CLS hook in the tracer was used.

Reply:
Right. Will try to improve this description.